test/jdk/java/util/Objects/BasicObjectsTest.java

```diff
@@ -306,4 +312,124 @@ private static int testNonNullOf() {
         }
         return errors;
     }
+
+    /**
+     * Test requireNonNull with null Supplier parameter.
+     * Should throw NullPointerException with null message (as per API spec).
+     */
+    private static int testRequireNonNullWithNullSupplier() {
+        int errors = 0;
+        try {
+            Objects.requireNonNull("test", (Supplier<String>) null);
+            // Should not reach here for non-null object
+            // But we're testing the supplier parameter
+            Objects.requireNonNull(null, (Supplier<String>) null);
+            System.err.println("testRequireNonNullWithNullSupplier: " +
+                            "failed to throw NPE");
+            errors++;
+        } catch (NullPointerException e) {
+            String msg = e.getMessage();
+            // According to Objects.java implementation, when messageSupplier is null,
+            // the exception message is also null
+            if (msg != null) {
+                System.err.println("testRequireNonNullWithNullSupplier: " +
+                                "expected null message but got: " + msg);
+                errors++;
+            }
+        }
+        return errors;
+    }
+
+    /**
+     * Test requireNonNull with Supplier that throws exception.
+     * The exception from supplier should be thrown, not wrapped.
+     */
+    private static int testRequireNonNullWithSupplierThrowingException() {
+        int errors = 0;
+        try {
+            Objects.requireNonNull(null, () -> {
+                throw new RuntimeException("Supplier exception");
+            });
+            System.err.println("testRequireNonNullWithSupplierThrowingException: " +
+                            "failed to throw exception");
+            errors++;
+        } catch (RuntimeException e) {
+            if (!"Supplier exception".equals(e.getMessage())) {
```

Comment:
I recommend creating the RuntimeException outside of the try block, and compare the identity of the caught exception.